### ISSUE TITLE
Troubles with Three.js importing (r90)

### ISSUE THREAD

**Issue by tokarev-i-v** [gregeryb/DAEBlend#6]
When I'm trying to import .dae scene to Three.js with ColladaLoader I have this Error
![2018-04-04 05 39 40](https://user-images.githubusercontent.com/10620596/38282637-a2e1c72c-37ca-11e8-849b-4463ae1543df.png)

There is a parameters:
![image](https://user-images.githubusercontent.com/10620596/38282659-c1b98b9e-37ca-11e8-8cd9-b51e0b581b0f.png)
Blender 2.75
Is there a way to solve this problem?

**Comment by gregdavisd** [comment]
Hello polyzer, It appears that the Three.js team has rewritten the Collada skinning code in the last few months so the 'Three.js' compatibility option may no longer be working. To look into it I would need the DAE file and details on the steps taken to reproduce the error. I don't know what the error means for Three.js. You could also ask the Three.js team if it's a known problem.

**Comment by tokarev-i-v** [comment]
Ok. I have this problem only using this DAEexporter. (Standard Blender exporter works fine)
[fox_sits.zip](https://github.com/gregeryb/DAEBlend/files/1875948/fox_sits.zip)

**Comment by gregdavisd** [comment]
Change the "neck top" and "neck down" bone names to remove the space. e.g. "neck_top" and "neck_down". See if that fixes anything.

**Comment by tokarev-i-v** [comment]
Unfortunetly... it doesn't.
![image](https://user-images.githubusercontent.com/10620596/38378499-5b4e6f84-3917-11e8-996d-a668e4e6088b.png)
But for time Error was changed to another, before I reload scene in Blender.
There is my .blend file, may be it will help you:
[Fox.zip](https://github.com/gregeryb/DAEBlend/files/1880896/Fox.zip)

**Comment by gregdavisd** [comment]
Hello polyzer, 

I've reproduced the problem and it is caused by the fox mesh being under, or parented by, the armature controller. The three.js loader encounters the controller under the skeleton and recurses again to load the skeleton which in turn finds the controller again and so on until a stack overflow occurs.

 The exported parenting setup is correct as per the Collada 1.5 specification, see page 4-9 of collada_spec_1_5.pdf. So the problem is with https://github.com/mrdoob/three.js.

Also there is other bugs in three.js like it putting an identity matrix in the skeleton if it encounters an unreferenced bone sometimes causing the geometry to fold up into a mess.

The three.js code for skinning attempts to rebuild the skeleton based on the bones listed in the controller. However this rebuild does not always faithfully recreate the skeleton. This tree rebuilding procedure is a variation from how skinning is designed in Collada. As per the Collada design the skeleton tree structure is already defined with nested the XML <node> tags. This structure should be left as it is.

In short, getting this exporter to work with three.js will require many more workarounds that I don't have the time to implement. The upcoming release of Blender 2.8 will require this exporter to be rewritten for the new Blender API so the new version may get a workaround. In the meantime consider this exporter as incompatible with three.js.

Regards,
Greg.
